tests/integration/constraints/test_tabular.py

```diff
@@ -0,0 +1,39 @@
+import numpy as np
+import pandas as pd
+
+from sdv.metadata import SingleTableMetadata
+from sdv.single_table import GaussianCopulaSynthesizer
+
+
+def test_fixed_combinations_with_nans():
+    """Test that FixedCombinations constraint works with NaNs."""
+    # Setup
+    data = pd.DataFrame({
+        'A': [1, 2, np.nan, 1, 2, 1],
```

Comment:
I don't get how this matches the issue if the issue said the column was an int? The nan will make this column a float

Reply:
The bug turns out to be bigger than what we thought. It's related to FixedCombinations not being able to map combinations with NaN correctly, especially with a column with integer dtype, which causes a crash. The integer column in the test is the column B.

To give you more context, here's what is happening during the test:
- During fitting, the constraint maps the combination (`np.nan`, 30) to something.
- However, in the transformer, since we can't compare `np.nan`, the combination (np.nan, 30) is mapped to `None`.
- The UniformEncoder then learns that there is None in the column and will generate `None` in the same proportion.
- When the constraint reverses the transformation, there is no mapping for None to a combination, resulting in the value being set to `np.nan` for all the columns of the combinations.
- Because B is an integer column, it can't have `np.nan`, causing the data processor to crash when attempting to cast the column back to its dtype.

To fix this issue, we need to change all `np.nan` in a combination to `None`, making the combination mappable. You can test this fix using the following code snippet on the `main` branch. It shouldn't crash this time because B is an object column but there will be no (`np.nan`, ’30’) combination in the synthetic data. 
```python
import numpy as np
import pandas as pd

from sdv.metadata import SingleTableMetadata
from sdv.single_table import GaussianCopulaSynthesizer

data = pd.DataFrame({
    'A': [1, 2, np.nan, 1, 2, 1],
    'B': ['10', '20', '30', '10', '20', '10'],
})
metadata = SingleTableMetadata().load_from_dict({
    'columns': {
        'A': {'sdtype': 'categorical'},
        'B': {'sdtype': 'categorical'},
    }
})

synthesizer = GaussianCopulaSynthesizer(metadata)
my_constraint = {
    'constraint_class': 'FixedCombinations',
    'constraint_parameters': {'column_names': ['A', 'B']},
}
synthesizer.add_constraints(constraints=[my_constraint])

# Run
synthesizer.fit(data)
synthetic_data = synthesizer.sample(1000)
```
Let me know if it makes sense

Reply:
That does make sense. Can we add a test with two int columns in the combination to sanity check the issue if such a test doesn't already exist?

Reply:
Yes, done in [617d5c4](https://github.com/sdv-dev/SDV/pull/2185/commits/617d5c4bfb78e7ee58bc81ea6fcf368faf6470a6)